### PR TITLE
Add cgal-cpp for pure C++ part of CGAL

### PR DESCRIPTION
**Pull request by tadeu**
The idea is to separate CGAL into `cgal` and `cgal-cpp`, just like Boost
is separated in `boost` and `boost-cpp`.

This is based on current `cgal-feedstock` recipe, with bindings removed
and some updates.

### CODE REVIEW

File: recipes/cgal-cpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{% set version = "4.13" %}
+
+package:
+  name: cgal
```
Comment (marcelotrevisani) [inline recipes/cgal-cpp/meta.yaml]: Please rename it to cgal-cpp

**Comment by tadeu** [comment]
I don't know why the linter says that a feedstock already exists with this name, I didn't find it. The feedback that exists is `cgal`, not `cgal-cpp`.

File: recipes/cgal-cpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  # mpfr only supports vc14
+  skip: true  # [win and py<35]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - eigen
+  host:
```
Comment (marcelotrevisani) [inline recipes/cgal-cpp/meta.yaml]: ``eigen`` should be in host

File: recipes/cgal-cpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    - python
+    - boost-cpp
+    - mpfr
+  run:
+    - python
```
Comment (marcelotrevisani) [inline recipes/cgal-cpp/meta.yaml]: If it's pure c++, why do you need python?

File: recipes/cgal-cpp/meta.yaml (hunk 1 above)
Comment (tadeu) [reply recipes/cgal-cpp/meta.yaml]: Ahh, it was this ;)

**Comment by tadeu** [comment]
Although it failed in Azure/osx, it passed in CircleCI, is this ok?

**Comment by tadeu** [comment]
Looks like a configuration issue in Azure/osx:

```
+ cmake .. '-GUnix Makefiles' -DCMAKE_BUILD_TYPE=Release -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON
CMake Error at /Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/share/cmake-3.14/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/x86_64-apple-darwin13.4.0-clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/test_tmp/test/build_default/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/make cmTC_8509e/fast 
    /Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/make -f CMakeFiles/cmTC_8509e.dir/build.make CMakeFiles/cmTC_8509e.dir/build
    make[1]: Entering directory '/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/test_tmp/test/build_default/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_8509e.dir/testCCompiler.c.o
    /Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/x86_64-apple-darwin13.4.0-clang   -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -I/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -fdebug-prefix-map=/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/test_tmp=/usr/local/src/conda/cgal-cpp-4.13 -fdebug-prefix-map=/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho=/usr/local/src/conda-prefix  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9   -o CMakeFiles/cmTC_8509e.dir/testCCompiler.c.o   -c /Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/test_tmp/test/build_default/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_8509e
    /Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/cmake -E cmake_link_script CMakeFiles/cmTC_8509e.dir/link.txt --verbose=1
    /Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/x86_64-apple-darwin13.4.0-clang -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -I/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -fdebug-prefix-map=${SRC_DIR}=/usr/local/src/conda/${PKG_NAME}-${PKG_VERSION} -fdebug-prefix-map=${PREFIX}=/usr/local/src/conda-prefix  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib -L/Users/vsts/agent/2.148.1/work/1/a/miniconda/conda-bld/cgal-cpp_1553169914486/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib  CMakeFiles/cmTC_8509e.dir/testCCompiler.c.o  -o cmTC_8509e 
    ld: warning: ignoring file /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd, file was built for unsupported file format ( 0x2D 0x2D 0x2D 0x20 0x21 0x74 0x61 0x70 0x69 0x2D 0x74 0x62 0x64 0x2D 0x76 0x33 ) which is not the architecture being linked (x86_64): /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd
    ld: dynamic main executables must link with libSystem.dylib for architecture x86_64
    clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation) 
```

(this passes in Travis CI)

anyway, I'll also add `{{compiler('c')}}` to test requirements, besides `cxx`, just in case.

**Comment by tadeu** [comment]
Looks good to go :D

**Pull request merged**